Allow schema to validate using Xerces-C

stevedlawrence opened the pull request:
Xerces-C has a bug where it cannot correctly validate choices with an empty sequnce as one of the branches. This replaces the empty sequence with an empty complex type, which xerces-C does support. This should also be portable for IBM DFDL. The downside is now the infoset always constains a NoStructureData element, though there is no workaround for this since Xerces-C needs an element in the choice.

stevedlawrence commented on src/main/resources/com/owlcyberdefense/syslog/xsd/syslog-rfc5424.dfdl.xsd:
```diff
@@ -149,20 +149,11 @@ SOFTWARE.
+          <xs:complexType>
+            <xs:sequence />
+          </xs:complexType>
         </xs:element>
```
Actually, ignore this change. I tested with IBM DFDL and it does not allow complex types with empty content. Need another alternative that can work with Xerces-C. The commented out `NoStructureData` with length 1 and an assertion that the parsed value is a hyphen may be the best approach.

mbeckerle replied:
Agree. 

Sticking with the IBM portable DFDL subset requires you to do this with a real element, string length 1 that must be a hyphen. 

I can't see any way to avoid it. 

You should use an explicit simple type so you can put an enumeration facet on it with value="-". That defends unparsing (so long as one validates before unparsing). It will also silence schema quality checkers like xsat2.

mbeckerle reviewed (approved):
With change to put back making the element be length 1 with an assert on its value being "-" and  an enumeration facet value="-".

mbeckerle reviewed (approved):
With change to put back making the element be length 1 with an assert on its value being "-" and  an enumeration facet value="-".

mbeckerle reviewed (approved):
With change to put back making the element be length 1 with an assert on its value being "-" and  an enumeration facet value="-".

mbeckerle reviewed (approved):
With change to put back making the element be length 1 with an assert on its value being "-" and  an enumeration facet value="-".

mbeckerle reviewed (approved):
With change to put back making the element be length 1 with an assert on its value being "-" and  an enumeration facet value="-".

stevedlawrence committed:
Allow schema to validate using Xerces-C

Xerces-C has a bug where it cannot correctly validate choices with an
empty sequnce as one of the branches. This replaces the empty sequence
with an element parsing a string of length one that ensures the value is
a hyphen, which xerces-C does support. The downside is now the infoset
always constains a NoStructureData element, though there is no
workaround for this since Xerces-C needs an actual element in the
choice.

This method has been tested to be portable with IBM also be portable for
IBM DFDL. Additionally, xsi:type attributes were added to TDML infosets
to account for differences with timestamps representations in the
infoset with IBM DFDL. And fixed the portable priority value type to
correctly include the namespace prefix.
